packages/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -3944,20 +3944,91 @@ export default class Meeting extends StatelessWebexPlugin {
    */
   getDevices = () => Media.getDevices();
 
+  /**
+   * Handles ROAP_FAILURE event from the webrtc media connection
+   *
+   * @param {Error} error
+   * @returns {void}
+   */
+  handleRoapFailure = (error) => {
+    const sendBehavioralMetric = (metricName, error, correlationId) => {
+      const data = {
+        code: error.code,
+        correlation_id: correlationId,
+        reason: error.message,
+        stack: error.stack
+      };
+      const metadata = {
+        type: error.cause?.name || error.name
+      };
+
+      Metrics.sendBehavioralMetric(metricName, data, metadata);
+    };
+
+
+    if (error instanceof MC.Errors.SdpOfferCreationError) {
+      sendBehavioralMetric(BEHAVIORAL_METRICS.PEERCONNECTION_FAILURE, error, this.id);
+
+      Metrics.postEvent({
+        event: eventType.LOCAL_SDP_GENERATED,
+        meetingId: this.id,
+        data: {
+          canProceed: false,
+          errors: [
+            Metrics.generateErrorPayload(2001, true, MetricsError.name.MEDIA_ENGINE)]
+        }
+      });
+    }
+    else if ((error instanceof MC.Errors.SdpOfferHandlingError) || (error instanceof MC.Errors.SdpAnswerHandlingError)) {
+      sendBehavioralMetric(BEHAVIORAL_METRICS.PEERCONNECTION_FAILURE, error, this.id);
```

Comment:
once you have some examples i can filter them out in amplititude

Reply:
you can already do this using the code - for example you can find one from today where I've triggered a fake setRemoteDescription failure by putting "assdfdfd" string instead of the sdp - you can find it by searching for "js_sdk_peerConnection_failures" with code=30004